### DISCUSSION TITLE
@Ctx() decorator default type

### DISCUSSION THREAD

**Discussion by matejanajdanov** [MichalLytek/type-graphql#1193] [Q&A] [answered]
Is there a way we can set a default type to the context so I don't have to explicitly type it every time I need to use it in resolvers?

example: 
```
const apolloServer = new ApolloServer({
      schema,
      plugins,
      context: ({ req, res }):MyContext => ({ req, res }),
    });


@Mutation(() => Response)
  async login(@Arg('credentials') credentials: LoginInput, @Ctx() { req }): Promise<Response> {
   console.log(req.headers)
  }
```

so `req.headers` isn't inferring the type from `MyContext` so I have to do this:

```
@Mutation(() => AuthResponse)
  async login(@Arg('credentials') credentials: LoginInput, @Ctx() { req }:MyContext ): Promise<Response> {
   console.log(req.headers)
  }
```

Is this really necessary and is there a way we can type it when we are declaring the context? 

Thank you in advance!! 🙏 🙏 🙏

**Comment by MichalLytek** [comment, marked as answer]
Yes, it's necessary. The same you can't set default type for args in TS config. It has to be explicit.
